ftw/tabbedview/statestorage.py

```diff
@@ -32,22 +32,25 @@ def get_key(self):
         key = []
         key.append('ftw.tabbedview')
 
-        # add the portal_type of the context
-        key.append(self.context.portal_type)
+        self._append_portal_type(key)
+        self._append_view_name(key)
+        self._append_userid(key)
 
-        # add the name of the tab
+        # concatenate with "-"
+        return '-'.join(key)
+
+    def _append_view_name(self, key):
         key.append(self.tabview.__name__)
 
+    def _append_portal_type(self, key):
+        key.append(self.context.portal_type)
+
+    def _append_userid(self, key):
         # add the userid
         mtool = getToolByName(self.context, 'portal_membership')
         member = mtool.getAuthenticatedMember()
-        if not member or member == AccessControl.getSecurityManager().getUser():
-            return None
-
-        key.append(member.getId())
-
-        # concatenate with "-"
-        return '-'.join(key)
+        if member and member != AccessControl.getSecurityManager().getUser():
```

Comment:
I'm not sure about this behavior change:
 - With the new implementation it will raise an AttributeError, if the request is anonymous or the current User can't be fetched. The former implementation just returns None.

@jone what do you think?